File: notification/java/src/main/java/com/nublic/notification/java/Notification.java
// Copyright (c) 2011, Nublic

/** 
 * @author David Navarro Estruch
 */

package com.nublic.notification.java;

import org.freedesktop.dbus.DBusConnection;
import org.freedesktop.dbus.exceptions.DBusException;



public class Notification {
	/**
	*
	* @author David Navarro Estruch
	*/
	private static String ObjectPath = "/com/nublic/notification/Messages";
	private static String ServiceBusName = "com.nublic.notification";
	private static DBusConnection conn;

	public void newMessage(String app, String user, String level, String text) {
		try {
			conn = DBusConnection.getConnection(DBusConnection.SYSTEM);
			MessageConnection c = (MessageConnection) conn.getRemoteObject(ServiceBusName, ObjectPath, MessageConnection.class);
			c.new_message(app, user, level, text);
		} catch(DBusException ex) {
			ex.printStackTrace();
		}
	}
}

File: python/client/src/nublic/notification.py
from dbus_in_other_thread import call_without_return
from rpcbd_client import rpcbd_call

def send_notification(app, user, level, text, use_dbus=False):
    if use_dbus:
        return call_without_return('com.nublic.notification',
                                   '/com/nublic/notification/Messages',
                                   'com.nublic.notification',
                                   lambda i: i.new_message(app, user, level, text))
    else:
        rpcbd_call(5441, lambda i: i.new_message(app, user, level, text))
